trl/trainer/callbacks.py

```diff
@@ -253,7 +253,8 @@ def on_train_begin(self, args: TrainingArguments, state: TrainerState, control:
         tokenizer = kwargs["tokenizer"]
         tokenizer.padding_side = "left"
         accelerator = self.trainer.accelerator
-        model = getattr(self.trainer, "ref_model", kwargs["model"])  # get the ref model if any, else use the model
+        # Use the reference model if available, otherwise use the initial model
+        model = getattr(self.trainer, "ref_model", self.trainer.model_wrapped)
```

Comment:
Does it work in the peft case? ie when ref_model is None?

Reply:
Oh good catch - let me see if I can write a unit test for this case

Reply:
Test added here with some minor cleanup to reuse the same expected winrates across tests: https://github.com/huggingface/trl/pull/2074/commits/592a0880dcd800e8cf2f42e3916f0a969f3f84fd

Reply:
OK to merge @qgallouedec or would you like some final changes?